src/platform/packages/private/kbn-grid-layout/grid/grid_panel/grid_panel.tsx

```diff
@@ -74,12 +75,24 @@ export const GridPanel = React.memo(({ panelId, rowId }: GridPanelProps) => {
         .pipe(skip(1)) // skip the first emit because the `initialStyles` will take care of it
         .subscribe(([interactionEvent, gridLayout, proposedGridLayout]) => {
           const ref = gridLayoutStateManager.panelRefs.current[rowId][panelId];
-          const panel = (proposedGridLayout ?? gridLayout)[rowId]?.panels[panelId];
+          const draggedPanel = proposedGridLayout?.[rowId]?.panels[panelId];
+          const confirmedPanel = gridLayout[rowId]?.panels[panelId];
+          const panel = draggedPanel ?? confirmedPanel;
           if (!ref || !panel) return;
 
           if (panelId === interactionEvent?.id) {
             ref.classList.add('kbnGridPanel--active');
 
+            // The `kbnGridPanel--activeModified` class disables pointer events when a panel is already being dragged and its position changed.
+            // We can't apply this to `kbnGridPanel--active` directly, as it would block regular clicks
+            // within the panel header since dragging starts on a `mousedown` event.
+            const isPanelModified = !isEqual(draggedPanel, confirmedPanel);
+            if (isPanelModified) {
+              ref.classList.add('kbnGridPanel--activeModified');
+            } else {
+              ref.classList.remove('kbnGridPanel--activeModified');
+            }
```

Comment:
I'm a bit confused why this changed, because this definitely used to work 😅 This feels overcomplicated to me, so I wonder if we can dig in a bit deeper and figure out what changed so that this is not necessary?

Reply:
Yes- we could instead revert merging activePanel and interactionEvent subscription. The problem is that when we merge these into one we apply kbnGridPanel--active class immediately on click and not on mousemove after the click. That makes all the other events within the panel unreachable since the class supplies 'pointer-events:none' styles. We could revert it and I considered it. 
However, apart from the fact that having these two subscriptions separately also adds complexity, there's one more advantage to this solution - it adds the proper styling from the start of interaction and not when the panel moved and it looks better for keyboard. 

To sum up - both old and new solutions come with extra complexity. I'm ok with both leaning slightly to the new one.

Reply:
I really think that efficiency **should be king** when it comes to this subscription, because it fires so often - everything we add here has potential to add lag to the dragging of panels, and considering that deep object comparison is expensive, I'm not a fan of adding it here. 

I am in favour of whatever route removes this comparison from this subscription :+1:

Reply:
> deep object comparison is expensive

Ok, I assumed that it isn't expensive, since it only contains 5 primitives (column, row, width, height and id). 

But I hear you and after reading your comments I feel like I am trying to solve a problem that can cause some other probelms in a PR that should be about keyboard navigation. Maybe merging these two subscriptions requires a deeper thought - for example, right now we run the `interactionEvent` panel subscription on a simple click, and before it was an `activePanel` subscription that would happen already when user had an intent to drag. I feel the most comfortably reverting my changes and implementing them separately 🫠

Reply:
Since the separation of `activePanel` and `interactionEvent` happened **in this PR** (and it is technically unrelated to the overall goal of supporting keyboard events), perhaps the simplest solution is just to revert that for now and tackle this cleanup as a separate PR so that this work can go forward?

Reply:
> I feel the most comfortably reverting my changes and implementing them separately 

LOL we came to the same conclusion at the same time 🤣 Sounds good to me!